Documentation/ceph-csi-drivers.md

```diff
@@ -175,3 +175,91 @@ CSI-Addons supports the following operations:
   - [Annotating PersistentVolumeClaims](https://github.com/csi-addons/kubernetes-csi-addons/blob/v0.3.0/docs/reclaimspace.md#annotating-perstentvolumeclaims)
 - Network Fencing
   - [Creating a NetworkFence](https://github.com/csi-addons/kubernetes-csi-addons/blob/v0.3.0/docs/networkfence.md)
+
+### Enable RBD Encryption Support
+
+Ceph-CSI supports encrypting RBD volumes with LUKS encryption. More details can be found
+[here](https://github.com/ceph/ceph-csi/blob/v3.5.1/docs/deploy-rbd.md#encryption-for-rbd-volumes)
+with full list of supported encryption configurations. A sample configmap can be found
+[here](https://github.com/ceph/ceph-csi/blob/v3.5.1/examples/kms/vault/kms-config.yaml).
```

Comment:
Thank you for updating the documentation. 

There are questions that end users might have that I don't see answered here or in the Ceph-CSI docs. It's important for documentation to explain for users what the benefits (and drawbacks) of configurations are, and I don't see that kind of information.

What benefits do users get from encrypting RBD volumes? How is this different from encrypting OSDs on-disk? If OSDs are encrypted on disk, is it still beneficial to enable this? What are the drawbacks of enabling encryption (I assume read/write throughput is reduced)?

Reply:
> Thank you for updating the documentation.
> 
> There are questions that end users might have that I don't see answered here or in the Ceph-CSI docs. It's important for documentation to explain for users what the benefits (and drawbacks) of configurations are, and I don't see that kind of information.

It's upto the users to determine largely which configuration they want to use. 
Various kinds on encryption types were added overtime, based on requirements at the time. 
I don't think we need to list such benefits or drawbacks or rather these are purely use case dependent. 
 
> What benefits do users get from encrypting RBD volumes? 

similar to benefits they get some osd encryption or any kind of encryption, data security. 

> How is this different from encrypting OSDs on-disk? 

I think it's implied that individual volumes may be encrypted with various other encryption kinds/keys.
Unlike in OSD case, where everything in a OSD is encrypted with same key.

>If OSDs are encrypted on disk, is it still beneficial to enable this? 

It's upto the users but this would lead to double encryption and obviously a performance hit.

>What are the drawbacks of enabling encryption (I assume read/write throughput is reduced)?

OSD encryption also suffers similar perf hit.
I think this is definitely obvious for users using encryption on any kind of storage.

Reply:
What seems obvious to you and to us as storage developers will not be obvious to all users. Many users want security but may not have the breadth and depth of understanding you are assuming they have. 

These docs need updates to help users decide whether this option to enable encryption is good for them. The documentation needs to clearly explain what is encrypted in this use-case. Given that there are also other encryption options in Rook, many users will be confused about how this is different from existing options (I know this because I myself have some confusion). Please do explain what exactly is encrypted and what is not. Also cross-link to other Rook's docs for on-disk encryption at minimum since that may be what some users actually want/need. Similarly, it might be good to cross-link here from a short note in the OSD encryption docs.

OSD configuration settings that give "encrypted" option are here: https://rook.io/docs/rook/v1.8/ceph-cluster-crd.html#osd-configuration-settings
I don't think we need to explicitly link to KMS-based encryption, but that is documented here: https://rook.io/docs/rook/v1.8/ceph-kms.html

Reply:
>There are questions that end users might have that I don't see answered here or in the Ceph-CSI docs. It's important for documentation to explain for users what the benefits (and drawbacks) of configurations are, and I don't see that kind of information.

I think this should be a doc request at ceph-csi repo. 

I tried to add some additional details and benefits that set this apart from OSD encryption.
Let me know how that sounds or if it needs any more changes.